core/src/main/java/jenkins/views/Header.java

```diff
@@ -54,4 +57,15 @@ public static Header get() {
         return header.orElseGet(JenkinsHeader::new);
     }
 
+    /**
+     * @return a list of {@link Action} to show in the header, defaults to {@link hudson.model.RootAction} extensions
+     */
+    @Restricted(NoExternalUse.class)
+    public List<Action> getActions() {
+        return Jenkins.get()
+                .getActions()
+                .stream()
+                .filter(e -> e.getIconFileName() != null)
+                .toList();
```

Comment:
```suggestion
                .sorted((x, y) -> {
                    if (x instanceof UserAction) return 1;
                    if (y instanceof UserAction) return -1;
                    return 0;
                })
                .toList();
```
Avoids that actions provided by newly installed plugins appear after the user action.
A bit hacky, ideally when new plugins are installed Jenkins takes care that the new extensions are added at the right place.

Reply:
Updated it to pull the ordinal each time, feels a little less hacky but still not ideal - feel it should be lower level.